### PR TITLE
Pagination changes given new vai cache backed steve api

### PR DESCRIPTION
**Pull request by richard-cox**
~Blocked on #10795~

### Summary
Contributes to https://github.com/rancher/dashboard/issues/9545 & https://github.com/rancher/dashboard/issues/9964

### Occurred changes and/or fixed issues
- Previous PRs worked with the existing api
- Changes in this PR are to support using the new vai backed api
  - fixed making requests where the page/pageSize are no needed
    - basically wherever we want to make use of the new filter mechanism
  - support both native array filtering (metadata.field.1) and new (metadata.field[1])
  - add validation for path used for sort (as per path used for filter)
- There are a number of sortable/filterable params that are waiting on a new backend build, 
  - params that aren't _yet_ supported but will in 2.9.0 have been added to the allow list in steve-pagination-utils
  - disabled sorting by State column, to enable in 2.9.0
  - disabled the workload type Health column. these rely on fetching all pods and then applying a label selector locally. Will be addressed via https://github.com/rancher/dashboard/issues/10417
- Ensure we use strict / exact filter params when we don't need partial matches
  - filtering via the list search text - partial matches ok
  - everywhere else - must be exact match
- Removed temp code which allowed enabling steve pagination without new feature flag

### Technical notes summary

- By the time this PR is ready to review the core steve cache changes should have merged
- See https://github.com/rancher/rancher/issues/45635 for pending backend changes
  - Note - api `count` is incorrect so ui will only show a single page of results and no page controls

### Areas or cases that should be tested
This assumes UI pointing to an instance containing the new steve cache changes.

With both new steve cache feature flag and pagination performance setting enabled, and with both off
- 2-3 random regression areas from https://github.com/rancher/dashboard/pull/10795
  - Note - for node/pod list pagination with server-side pagination enabled some columns will fail, see changes to `VALID_FIELDS`)
- In addition, if not done above
  - Node list shows populated CPU, RAM and Pods column
  - one secret selection (can be any secret based one from https://github.com/rancher/dashboard/pull/10795)

Specifically with steve cache feature flag and pagination performance setting enabled
- Workload based pages like deployments and jobs DON'T display the Health column
- Any list that uses the new pagination process 
  - filtering by text works in a partial way (entering `a` returns all rows with name containing `a`, rather than just rows with a name of `a`)

Specifically with steve cache feature flag and pagination performance setting disabled
- Workload based pages like deployments and jobs DO display the Health column

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Comment by richard-cox** [comment]
@codyrancher , @nwmac next PR ready to go. Smaller than the previous ones. One PR after this left, which is smaller again (and specifically to do with loading indicators)

File: shell/components/form/SelectOrCreateAuthSecret.vue [hunk 1]
```diff
@@ -411,9 +411,8 @@ export default {
       const findPageArgs = {
         // Of type ActionFindPageArgs
         namespaced: this.filterByNamespace ? this.namespace : '',
-        pagination: new PaginationArgs({
-          pageSize: -1,
-          filters:  [
+        pagination: new PaginationFilterArgs({
```
Comment (richard-cox) [inline shell/components/form/SelectOrCreateAuthSecret.vue:414]: - PaginationArgs
- PaginationFilterArgs --> FilterArgs

File: pkg/harvester-manager/machine-config/harvester.vue [hunk 2]
```diff
@@ -135,9 +135,9 @@ export default {
         let configMapsUrl = `${ url }/${ CONFIG_MAP }s`;
 
         if (this.$store.getters[`cluster/paginationEnabled`](CONFIG_MAP)) {
-          const pagination = new PaginationArgs({
-            page:     1,
-            pageSize: -1,
+          const pagination = new PaginationFilterArgs({
+            page:     null,
```
Comment (richard-cox) [inline pkg/harvester-manager/machine-config/harvester.vue:139]: no page, pageSize

File: shell/plugins/steve/steve-pagination-utils.ts [hunk 3]
```diff
@@ -116,13 +120,30 @@ class StevePaginationUtils extends NamespaceProjectFilters {
...
-      { field: 'metadata.labels' }
+      { field: 'metadata.labels[harvesterhci.io/cloud-init-template]' }
+    ],
+    [NAMESPACE]: [
+      { field: 'metadata.labels[field.cattle.io/projectId]' }
     ]
   }
 
+  private convertArrayPath(path: string): string {
+    if (path.startsWith('metadata.fields.')) {
```
Comment (richard-cox) [inline shell/plugins/steve/steve-pagination-utils.ts:134]: string length of `metadata.fields.`, sub string

File: shell/plugins/steve/steve-pagination-utils.ts [hunk 4]
```diff
@@ -132,9 +132,7 @@ class StevePaginationUtils extends NamespaceProjectFilters {
 
   private convertArrayPath(path: string): string {
     if (path.startsWith('metadata.fields.')) {
-      const [part1, part2, ...rest] = path.split('.');
-
-      return `${ part1 }.${ part2 }[${ rest.join('.') }]`;
+      return `metadata.fields[${ path.substring(16, path.length) }]`;
```
Comment (richard-cox) [inline shell/plugins/steve/steve-pagination-utils.ts:135]: remove path.length

**Review by nwmac — DISMISSED** [review]
Reviewed with RC

**Review by nwmac — APPROVED** [review]

**Pull request merged**